java/ql/lib/semmle/code/java/frameworks/spring/SpringHttp.qll

```diff
@@ -190,6 +190,11 @@ private class SpringXssSink extends XSS::XssSink {
         // String.
         returnType instanceof RawClass
       )
+      or
+      exists(RefType returnType |
+        returnType = requestMappingMethod.getReturnType() and
+        returnType.hasQualifiedName("org.owasp.webgoat.assignments", "AttackResult")
+      )
```

Comment:
There must be some other and more generic way to capture this. Somehow Spring is able to interprete `AttackResult` class instances in a generic way, so we should too. @atorralba could you advise?

Reply:
My understanding is that those things often get Jsonified or Xmlified or however it is configured. I guess the question is then how the data is interpreted on the other side with that side being, for example, JavaScript. I reckon it might cause trouble to say that all tainted return values will cause an XSS, when that might really depend on how the JS code uses the data. [In the case of WebGoat](https://github.com/zbazztian/webgoat-delombok/blob/9c868dcaab788549f1b0207e48f654bfce106f11/webgoat-container/src/main/resources/static/js/goatApp/view/LessonContentView.js#L124-L125) it looks like they don't do any filtering of that data, [other than removing some JSON characters](https://github.com/zbazztian/webgoat-delombok/blob/9c868dcaab788549f1b0207e48f654bfce106f11/webgoat-container/src/main/resources/static/js/goatApp/view/LessonContentView.js#L184)

Reply:
@atorralba Thanks for the detailed explanation, I agree! I will update the PR to only contain the taint steps we identified as generic enough for the standard library.